todo_app/templates/index.html

```diff
@@ -71,18 +70,93 @@ <h2>Items</h2>
             {% if view_model.sort_col == "target_date" and view_model.sort_dir == 'des' %}&#8593;  {% endif %} 
             </a></th>
      
+            <th scope="Col"><a href = {{ url_for('index', sort_col="last_activity_date", last_sort_col=last_sort_col, sort_dir=sort_dir) }}>Last Activity
+              {% if view_model.sort_col == "last_activity_date" and view_model.sort_dir == 'asc' %} &#8595;  {% endif %}
+              {% if view_model.sort_col == "last_activity_date" and view_model.sort_dir == 'des' %}&#8593;  {% endif %} 
+              </a></th>
+                  
+
             <th scope="Col">Action</th>
           </tr>
         </thead>
+        <tbody>
+          {% for item in view_model.get_items("To Do") %}
+            <tr>
+              <td>{{ item.id }}</td>
+              <td>{{ item.title }}</td>   
+              <td>{{ item.status }}</td>
+              <td>{{ item.target_date }}</td>
+              <td>{{ item.last_activity_date }}</td>
+              <td>
+                  {% if item.status=="To Do" %}
+                    <a href = {{ url_for('update_task', action="start_task", id_long=item.id_long, last_sort_col=last_sort_col, sort_dir=sort_dir)}}>[Start Task]</a>
+                  {% endif %}
+                  {% if item.status=="Doing" %}
+                    <a href = {{ url_for('update_task', action="finish_task", id_long=item.id_long, last_sort_col=last_sort_col, sort_dir=sort_dir)}}>[Finish Task]</a>
+                  {% endif %}
+                  {% if item.status=="Done" %}
+                    <a href = {{ url_for('update_task', action="reset_task", id_long=item.id_long, last_sort_col=last_sort_col, sort_dir=sort_dir)}}>[Reset Task]</a>
+                  {% endif %}
```

Comment:
Now that you've separated out the statuses with the ViewModel, I think we can simplify this considerably!
```suggestion
                  <a href = {{ url_for('update_task', action="start_task", id_long=item.id_long, last_sort_col=last_sort_col, sort_dir=sort_dir)}}>[Start Task]</a>
```

And similarly for the other statuses!